onnxruntime/contrib_ops/cuda/bert/group_query_attention.cc

```diff
@@ -108,72 +107,85 @@ Status GroupQueryAttention<T>::ComputeInternal(OpKernelContext* context) const {
   size_t softmax_lse_bytes = 0;
   size_t softmax_lse_accum_bytes = 0;
   size_t out_accum_bytes = 0;
-  size_t seqlens_k_bytes = 0;
   if (use_flash_attention) {
     // softmax buffer
     softmax_lse_bytes = onnxruntime::flash::get_softmax_lse_size(parameters.sequence_length, parameters.batch_size, parameters.num_heads);
     // split kv buffer
     using namespace std;
     auto [num_splits, slse_accum_bytes, o_accum_bytes] = onnxruntime::flash::get_num_splits_and_buffer_sizes(
-        parameters.batch_size, parameters.sequence_length, parameters.kv_sequence_length, parameters.num_heads,
+        parameters.batch_size, parameters.sequence_length, parameters.sequence_length, parameters.num_heads,
```

Comment:
>sequence_length [](http://example.com/codeflow?start=70&length=15)

is it mask_sequence_length?

Reply:
i believe it is the normal one

Reply:
It shall be the present sequece length. The idea is to increase the number of block by dividing on K and V:
[https://github.com/Dao-AILab/flash-attention/blob/83aef842beec1037eb8c1d9c3ef3ed8aae80b091/csrc/flash_attn/src/flash_fwd_launch_template.h#L76](https://github.com/Dao-AILab/flash-attention/blob/83aef842beec1037eb8c1d9c3ef3ed8aae80b091/csrc/flash_attn/src/flash_fwd_launch_template.h#L76)